jetty-ee10/jetty-ee10-maven-plugin/src/main/java/org/eclipse/jetty/ee10/maven/plugin/OverlayManager.java

```diff
@@ -123,17 +127,18 @@ protected List<Overlay> getOverlays()
      * @return the location to which it was unpacked
      * @throws IOException if there is an IO problem
      */
-    protected  Resource unpackOverlay(Overlay overlay)
+    protected Resource unpackOverlay(Overlay overlay)
         throws IOException
     {
         if (overlay.getResource() == null)
             return null; //nothing to unpack
 
         //Get the name of the overlayed war and unpack it to a dir of the
         //same name in the temporary directory
-        String name = overlay.getResource().getFileName();
+        Path p = Paths.get(URIUtil.unwrapContainer(overlay.getResource().getURI()));
+        String name = p.getName(p.getNameCount() - 1).toString();
```

Comment:
I don't think this is correct either, as if the overlay itself is a CombinedResource, then getURI will return null.    You are probably better just naming them "dir0", "dir1", ....
Or we can add a Resource.getSimpleName() method to extra a simple name that is not necessarily a file name.